lib/rspec/mocks/message_expectation.rb

```
@@ -581,6 +581,7 @@ def raise_already_invoked_error_if_necessary(calling_customization)
         end
 
         def set_expected_received_count(relativity, n)
+          return if negative?
```

Ideally it should warn when this happens...

---

Or should it raise an error?  Consider that it's invalid to set an expected count on a negative expectation, and has never worked (as demonstrated by #1210).

---

Would be equally happy in that scenario too.